common/types/block.go

```diff
@@ -105,6 +106,42 @@ func (id NodeID) ShortString() string {
 	return Shorten(name, 5)
 }
 
+// BytesToNodeID deserializes a byte slice into a NodeID
+func BytesToNodeID(b []byte) NodeID {
+	if len(b) < 32 {
+		var emptySlice []byte
+		return NodeID{
+			Key:          "",
+			VRFPublicKey: emptySlice,
+		}
+	}
+	pubKey := b[0:32]
+	vrfKey := b[32:]
+	return NodeID{
+		Key:          util.Bytes2Hex(pubKey),
+		VRFPublicKey: vrfKey,
+	}
+}
+
+//StringToNodeID deserializes a string into a NodeID
+func StringToNodeID(s string) (NodeID, error) {
+	strLen := len(s)
+	if strLen < 64 {
+		var emptySlice []byte
+		return NodeID{
```

Comment:
Why not just `return nil, err`?

Reply:
The `nil` was giving compile-time errors because it isn't of type NodeID, and can't be cast to a NodeID as well, so I just decided to return an empty NodeID struct